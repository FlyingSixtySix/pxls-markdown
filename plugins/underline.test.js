const { processor, expectProcessorParse } = require('../testUtil.js');
const underline = require('./underline.js');

processor.use(underline);

test(
	'tokenizes "__underline text__" correctly',
	() => expectProcessorParse('__underline text__').toMatchSnapshot()
);
test(
	'tokenizes "___underline and italic text___" correctly',
	() => expectProcessorParse('___underline and italic text___').toMatchSnapshot()
);
test(
	'tokenizes "____" correctly',
	// The "some text" padding is needed to avoid parsing thematic break
	() => expectProcessorParse('some text ____').toMatchSnapshot()
);
test(
	'tokenizes "__ __" correctly',
	// The "some text" padding is needed to avoid parsing thematic break
	() => expectProcessorParse('some test __ __').toMatchSnapshot()
);
test(
	'tokenizes "__\\n__" correctly',
	() => expectProcessorParse('__\n__').toMatchSnapshot()
);
