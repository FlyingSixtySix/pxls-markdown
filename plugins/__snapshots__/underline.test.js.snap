// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokenizes "__ __" correctly 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "type": "text",
        "value": "some test __ __",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`tokenizes "__\\n__" correctly 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "type": "text",
        "value": "__
__",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`tokenizes "____" correctly 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "type": "text",
        "value": "some text ____",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`tokenizes "___underline and italic text___" correctly 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "type": "text",
                "value": "underline and italic text",
              },
            ],
            "type": "emphasis",
          },
        ],
        "type": "underline",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`tokenizes "__underline text__" correctly 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "type": "text",
            "value": "underline text",
          },
        ],
        "type": "underline",
      },
    ],
    "type": "paragraph",
  },
]
`;
